File: SmartX.Client/src/models/sensor.ts
export type SensorCategory =
  | 'Environmental'
  | 'PowerConsumption'
  | 'Actuator'

export interface Sensor {
  id: string
  deviceIdentifier: string
  deploymentLocation: string
  room: string
  zone: string
  nodeId: string
  category: SensorCategory
  createdAtUtc: string
  updatedAtUtc: string
}

export interface SensorRequest {
  deviceIdentifier: string
  deploymentLocation: string
  room: string
  zone: string
  nodeId: string
  category: SensorCategory
}

File: SmartX.Api/Models/Sensor.cs
namespace SmartX.Api.Models
{
    public class Sensor
    {
        public Guid Id { get; set; }

        public string DeviceIdentifier { get; set; } = string.Empty;

        public string DeploymentLocation { get; set; } = string.Empty;

        public string Room { get; set; } = string.Empty;

        public string Zone { get; set; } = string.Empty;

        public string NodeId { get; set; } = string.Empty;

        public SensorCategory Category { get; set; }

        public DateTime CreatedAtUtc { get; set; }

        public DateTime UpdatedAtUtc { get; set; }
    }
}